src/components/Reports/ReportsHoursVisits/ReportsHoursVisits.js

```diff
@@ -51,11 +52,16 @@ const ReportsHoursVisits = ({ domainName, dateFrom, dateTo, dependencies: { data
             accumulator[weekDay][hour].withEmail += item.withEmail;
             accumulator[weekDay][hour].withoutEmail += item.withoutEmail;
           }
+          if (item.quantity > max) {
+            max = item.quantity;
```

Comment:
_reduce_ should not have side effects.

What do you think about adding a new property to the accumulator with the maximum value?

```js
        const visitsByWeekDayAndHour = hoursVisitsdata.value.reduce((accumulator, item) => {
          const weekDay = item.from.getUTCDay();
          const hour = item.from.getUTCHours();
          accumulator[weekDay][hour].quantity += item.quantity;
          accumulator.max = item.quantity > accumulator.max ? item.quantity : accumulator.max;
          // . . .
```

Reply:
The name `visitsByWeekDayAndHour` would be wrong, maybe rename it as `processedVisits` or something like that?

You can also change the structure, for example:

```
{
    byWeekDayAndHour: [
        // . . . 
    ], [
        // . . . 
    ], 
    // . ..
    max: 50
}
```

Reply:
Better example:

```js
const processedVisits = hoursVisitsdata.value.reduce((accumulator, item) => {
  // . . .
}, { byWeekDayAndHour: createEmptyWeekDayHoursMatrix(), max: 0 });
```